### PR TITLE
Make sure that getChildClient() is cheap

### PR DESCRIPTION
**Pull request by Raynos**
This makes two changes: first fix any missing options not
passed to child.

Secondly ensure that all children share the same dnsResolver.

r: @Matt-Esch @syyang

### CODE REVIEW

**Comment by Matt-Esch** [comment]
You can make it cheaper if create a new constructor function with parent as proto, and override this.prefix

**Comment by Raynos** [comment]
Changing proto messes with the hidden class. 

I can optimize the constructor further to make it cheaper but that seems like a micro-op. I just want to fix the shared DNSResolver lulz.

**Comment by syyang** [comment]
lgtm. thanks!

**Comment by Raynos** [comment]
before:

```
raynos at raynos-ThinkPad-T440p  ~/uber/node-statsd-client on fast-get-child-client*
$ node benchmarks/child-client.js 
time:  989
raynos at raynos-ThinkPad-T440p  ~/uber/node-statsd-client on fast-get-child-client*
$ node benchmarks/child-client.js 
time:  991
raynos at raynos-ThinkPad-T440p  ~/uber/node-statsd-client on fast-get-child-client*
$ node benchmarks/child-client.js 
time:  959
```

image


after:

```
$ node benchmarks/child-client.js 
time:  648
raynos at raynos-ThinkPad-T440p  ~/uber/node-statsd-client on fast-get-child-client*
$ node benchmarks/child-client.js 
time:  657
raynos at raynos-ThinkPad-T440p  ~/uber/node-statsd-client on fast-get-child-client*
$ node benchmarks/child-client.js 
time:  627
```

image


I made it 66% faster with the latest commit. 
Looking at the flames I think I made v8 inline all the constructors and stuff.

If it needs to be any faster i'm going to have to read the generated ASM:

File: lib/statsd-client.js [hunk 1]
```diff
@@ -8,41 +26,36 @@ var Messages = require('./messages.js');
...
-    // Prefix?
-    if (this.options.prefix && this.options.prefix !== '') {
-        // Add trailing dot if it's missing
-        var p = this.options.prefix;
-        this.options.prefix = p[p.length - 1] === '.' ?
-            p : p + '.';
-    }
+function createStatsDClient(options) {
+    options = options || {};
+    options.prefix = options.prefix || '';
```
Comment (Matt-Esch) [inline lib/statsd-client.js]: look how it mutates a foreign object ;_;

File: lib/statsd-client.js [hunk 2]
```diff
@@ -8,41 +26,36 @@ var Messages = require('./messages.js');
...
-    // Ephemeral socket
-    this._ephemeralSocket = this.options._ephemeralSocket ||
+    var _ephemeralSocket = (options && options._ephemeralSocket) ||
         new EphemeralSocket(options);
 
-    if (this.options.dnsResolver &&
-        this._ephemeralSocket.resolveDNS
+    if (options.dnsResolver &&
+        _ephemeralSocket.resolveDNS &&
+        !_ephemeralSocket._dnsResolver
```
Comment (Matt-Esch) [inline lib/statsd-client.js]: sneaky patch hiding in your perf

File: lib/statsd-client.js (hunk 1 above)
Comment (Raynos) [inline lib/statsd-client.js]: back-compat.

dispatch literally relies on shared mutation.

File: lib/statsd-client.js (hunk 2 above)
Comment (Raynos) [inline lib/statsd-client.js]: look at the commits. this change was done in a different commit.
